crypto/bio/bss_mem.c

```diff
@@ -62,6 +69,10 @@ static const BIO_METHOD secmem_method = {
 typedef struct bio_buf_mem_st {
     struct buf_mem_st *buf;   /* allocated buffer */
     struct buf_mem_st *readp; /* read pointer */
+    struct buf_mem_dgram_st *dgrams; /* dgram data */
```

Comment:
`STACK_OF(struct buf_mem_dgram_st)` would avoid re-inventing a data structure -- I know performance is the reason here.  Better would be a generic queue implementation or even fixing our stack to be a double ended queue...

Reply:
I have reimplemented it as a linked list.